### PR TITLE
SDT-43: CCD - Support case creation on behalf of a different user

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SDT-43

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: charts/civil-sdt/values.preview.template.yaml [hunk 1]
```diff
@@ -15,7 +15,14 @@ java:
     civil:
```
Comment (paulridout) [inline charts/civil-sdt/values.preview.template.yaml:15]: Key vault name should be civil-sdt.  The main branch has the correct name, so I think this branch needs to be updated with the changes from main.

File: charts/civil-sdt/values.preview.template.yaml [hunk 2]
```diff
@@ -15,7 +15,14 @@ java:
...
       secrets:
-        - name: AppInsightsInstrumentationKey
+        - name: civil-sdt-service-s2s-secret
+          alias: CIVIL_SDT_S2S_KEY
+        - name: civil-sdt-POSTGRES-USER
+          alias: SDT_DB_USERNAME
+        - name: civil-sdt-POSTGRES-PASS
+          alias: SDT_DB_PASSWORD
+        - name: civil-sdt-servicebus-connection-string
+          alias: CIVIL_SDT_SERVICEBUS_CONNECTION_STRING
```
Comment (paulridout) [inline charts/civil-sdt/values.preview.template.yaml:25]: Are these needed?  They are being set to the same as in values.yaml, so should be picked up from there.

File: charts/civil-sdt/values.preview.template.yaml [hunk 3]
```diff
@@ -15,7 +15,14 @@ java:
...
   secrets:
     CIVIL_SDT_SERVICEBUS_CONNECTION_STRING:
       secretRef: servicebus-secret-queue-{{ .Release.Name }}-servicebus-civil-sdt-in-out
```
Comment (paulridout) [inline charts/civil-sdt/values.preview.template.yaml:28]: Servicebus values have been changed in main branch (for servicebus v1.0.4).  Another indication that this branch needs to be updated with changes from main branch!

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/CMCApi.java [hunk 4]
```diff
@@ -27,6 +31,12 @@ public interface CMCApi {
 
     String SDT_REQUEST_ID = "SDTREQUESTID";
 
+    String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+
+    String FROM_DATE_TIME = "fromDateTime";
+
+    String TO_DATE = "toDate";
```
Comment (paulridout) [inline consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/CMCApi.java:38]: Should this be called TO_DATE_TIME?  I believe the to date has a time component as well.

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/impl/JudgementWarrantService.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    public JudgementWarrantResponse judgementWarrantRequest(String authorization,
+                                                            String serviceAuthorization,
+                                                            String idamId,
+                                                            String sdtRequestId,
+                                                            JudgementWarrantRequest judgementWarrantRequest) {
+        return cmcApi.judgementWarrantRequest(authorization,
+                                              serviceAuthorization,
+                                              idamId,
+                                              sdtRequestId, judgementWarrantRequest
+        );
```
Comment (paulridout) [inline consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/impl/JudgementWarrantService.java:30]: Does this need to be on a new line? Methods in CMCApiFallback have it at the end of the previous line.

File: config/owasp/suppressions.xml [hunk 6]
```diff
@@ -22,10 +22,12 @@
       CVE-2023-20862
       CVE-2023-20863
       CVE-2023-20873
+      CVE-2023-24998
     </notes>
     <cve>CVE-2023-20862</cve>
     <cve>CVE-2023-20863</cve>
     <cve>CVE-2023-20873</cve>
+    <cve>CVE-2023-24998</cve>
```
Comment (paulridout) [inline config/owasp/suppressions.xml:30]: Check these again after updating branch with changes from main branch.  Hopefully all of them have been resolved.

File: consumers/src/main/java/uk/gov/moj/sdt/consumers/config/ConsumersConfig.java [hunk 7]
```diff
@@ -42,32 +37,17 @@ public class ConsumersConfig {
...
+                                                                                      @Qualifier("CacheSetupOutboundInterceptor")
+                                                                                      CacheSetupOutboundInterceptor cacheSetupOutboundInterceptor,
+                                                                                      @Qualifier("CacheEndOutboundInterceptor")
+                                                                                          CacheEndOutboundInterceptor cacheEndOutboundInterceptor,
+                                                                                      @Qualifier("XmlInboundInterceptor")
+                                                                                          XmlInboundInterceptor xmlInboundInterceptor,
+                                                                                      @Qualifier("SdtUnmarshallInterceptor")
+                                                                                          SdtUnmarshallInterceptor sdtUnmarshallInterceptor
+                                                                                      ) {
```
Comment (paulridout) [inline consumers/src/main/java/uk/gov/moj/sdt/consumers/config/ConsumersConfig.java:50]: I think the Qualifier annotations require the corresponding classes to be annotated with Component, but they don't seem to be, e.g. the XMLOutboundInterceptor class isn't annotated with `@Component("XmlOutboundInterceptor")`.  The main branch has this change in, plus the annotated classes.  I think this branch needs to be updated with the changes from the main branch.

File: consumers/src/unit-test/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        mockJudgementWarrantService = new MockJudgementWarrantService();
+    }
+
+    @Test
+    void getClient() {
+        JudgementWarrantRequest request = mock(JudgementWarrantRequest.class);
+        JudgementWarrantResponse response = mockJudgementWarrantService.judgementWarrantRequest(AUTHORISATION,
+                                                                                                SERVICE_AUTHORISATION,
+                                                                                                IDAM_ID_HEADER,
+                                                                                                SDT_REQUEST_ID , request);
```
Comment (paulridout) [inline consumers/src/unit-test/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantServiceTest.java:41]: Could put the request parameter on a new line.

File: infrastructure/key-vault.tf [hunk 9]
```diff
@@ -8,16 +8,17 @@ data "azurerm_key_vault" "s2s_vault" {
...
+
+data "azurerm_key_vault_secret" "key_from_vault" {
+  name         = "microservicekey-civil-sdt"
+  key_vault_id = data.azurerm_key_vault.s2s_vault.id
+}
+
+resource "azurerm_key_vault_secret" "s2s" {
+  name         = "civil-sdt-service-s2s-secret"
+  value        = data.azurerm_key_vault_secret.key_from_vault.value
+  key_vault_id = data.azurerm_key_vault.civil_vault.id
```
Comment (paulridout) [inline infrastructure/key-vault.tf:20]: Key vault has changed to civil-sdt specific vault.  Update this branch with latest changes from main branch and change key_vault_id value to `module.civil_sdt_key_vault.key_vault_id`

File: charts/civil-sdt/values.preview.template.yaml (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply charts/civil-sdt/values.preview.template.yaml:25]: this is just a template file,

**Pull request merged**